mlos_core/mlos_core/optimizers/observations.py

```diff
@@ -0,0 +1,252 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+"""Contains classes for observations and suggestions."""
+
+from __future__ import annotations
+
+from typing import Any, Iterator, Optional
+
+import pandas as pd
+
+
+def compare_optional_dataframe(
+    left: Optional[pd.DataFrame], right: Optional[pd.DataFrame]
```

Comment:
```suggestion
    left: Optional[pd.DataFrame], right: Optional[pd.DataFrame],
```
nit: trailing comma and black reformat